big_vision/trainers/proj/gsam/train.py

```diff
@@ -0,0 +1,401 @@
+# Copyright 2022 Big Vision Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Training loop example.
+This is a basic variant of a training loop, good starting point for fancy ones.
+"""
+from functools import partial
+import importlib
+import multiprocessing
+import os
+
+from absl import app
+from absl import flags
+from absl import logging
+import big_vision.evaluators.common as eval_common
+import big_vision.input_pipeline as input_pipeline
+import big_vision.optax as bv_optax
+import big_vision.pp.builder as pp_builder
+import big_vision.utils as u
+from clu import parameter_overview
+import flax
+import flax.jax_utils as flax_utils
+import jax
+import jax.config
+import jax.nn
+import jax.numpy as jnp
+import jax.profiler
+from ml_collections import config_flags
+import numpy as np
+import optax
+import tensorflow as tf  # pylint: disable=unused-import (for ThreadPool, weird)
+import tensorflow.io.gfile as gfile
+from gsam import gsam_gradient
```

Comment:
This should rather look like `from big_vision.trainers.proj.gsam.gsam import gsam_gradient`, and then move higher up in the correct position.